src/requests/textdocument.jl

```diff
@@ -59,10 +52,20 @@ function process(r::JSONRPC.Request{Val{Symbol("textDocument/didClose")},DidClos
     doc = getdocument(server, URI2(uri))
     empty!(doc.diagnostics)
     publish_diagnostics(doc, server)
-    if !is_workspace_file(doc)
-        deletedocument!(server, URI2(uri))
-    else
+    if is_workspace_file(doc)
         set_open_in_editor(doc, false)
+    else
+        if any(d.root == doc.root && doc._open_in_editor for (uri, d::Document) in getdocuments_pair(server) if d != doc)
```

Comment:
```suggestion
        if any(d.root == doc.root for d::Document in getdocuments_value(server) if d != doc)
```
I think we don't need to check for `doc._open_in_editor`, we will only get this message for docs that _are_ open in the editor by definition, right?

Reply:
Ooops, this is a typo, should read `d._open_in_editor` which is actually consistent with my comment within the file...

Changing the above addresses your comment below